sdk/cosmosdb/cosmos/test/public/functional/itemIdEncoding.spec.ts

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import assert from "assert";
+import { Suite } from "mocha";
+import { Container } from "../../../src";
+import { getTestContainer, removeAllDatabases } from "../common/TestHelpers";
+
+interface ItemPayload {
+  id?: string;
+  pk?: string;
+}
+
+interface TestScenario {
+  name: string;
+  id: string;
+  expectedCreateStatusCode: number;
+  expectedReadStatusCode?: number;
+  expectedReplaceStatusCode?: number;
+  expectedDeleteStatusCode?: number;
+  expectedCreateErrorMessage?: string;
+}
+
+const createPayload = function (id: string): ItemPayload {
+  return {
+    id: id,
+    pk: id,
+  };
+};
+
+const executeTestCase = async function (scenario: TestScenario) {
+  const container: Container = await getTestContainer(scenario.name, undefined, {
+    partitionKey: {
+      paths: ["/pk"],
+      version: undefined,
+    },
+    throughput: 25100,
+  });
+
+  try {
+    const response = await container.items.create(createPayload(scenario.id));
+    assert.strictEqual(response.statusCode, scenario.expectedCreateStatusCode);
+    assert.strictEqual(response.item.id, scenario.id);
+    if (response.resource) {
+      assert.strictEqual(response.resource.id, scenario.id);
+      assert.strictEqual(response.resource.pk, scenario.id);
+    } else {
+      assert.fail("response.resource should not be null");
+    }
+  } catch (err: any) {
+    if (err.code) {
```

Comment:
same as above! or can merge in single line as if (err.code && err.code === 'ERR_ASSERTION')

Reply:
Added err null-check